### PR TITLE
add support for binance_delivery api (dapi)

### PR DESCRIPTION
**Pull request by txu2014**
* Feature: new exchange: Binance Delivery.
### Description of code - what bug does this fix / what feature does this add?

- [ ] - Tested
- [ ] - Changelog updated
- [ ] - Contributors file updated (optional)

### CODE REVIEW

**Comment by bmoscon** [comment]
please undo the formatting changes made that shorten lines you didn't touch.

**Review by bmoscon — CHANGES_REQUESTED** [review]
undo formatting changes to un-modified code

**Comment by txu2014** [comment]
> undo formatting changes to un-modified code

removed duplicated lines caused by line wrapping.

File: cryptofeed/exchange/binance_delivery.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Please see the LICENSE file for the terms and conditions
+associated with this software.
+'''
+from decimal import Decimal
+import logging
+
+from yapic import json
+
+from cryptofeed.defines import BINANCE_FUTURES, BINANCE_DELIVERY, OPEN_INTEREST, TICKER
```
Comment (bmoscon) [inline cryptofeed/exchange/binance_delivery.py:12]: binance_futures import is not used in this file, please remove

File: cryptofeed/pairs.py [hunk 2]
```diff
@@ -307,18 +312,22 @@ def deribit_pairs():
     data = []
     for c in currencies:
         for k in kind:
-            data.extend(requests.get(f"https://www.deribit.com/api/v2/public/get_instruments?currency={c}&expired=false&kind={k}").json()['result'])
```
Comment (bmoscon) [inline cryptofeed/pairs.py:310]: undo formatting

File: cryptofeed/pairs.py [hunk 3]
```diff
@@ -222,7 +226,8 @@ def huobi_dm_pairs():
     for e in r['data']:
         pairs[f"{e['symbol']}_{mapping[e['contract_type']]}"] = e['contract_code']
         _exchange_info[HUOBI_DM]['tick_size'][e['contract_code']] = e['price_tick']
-        _exchange_info[HUOBI_DM]['short_code_mappings'][f"{e['symbol']}_{mapping[e['contract_type']]}"] = e['contract_code']
+        _exchange_info[HUOBI_DM]['short_code_mappings'][f"{e['symbol']}_{mapping[e['contract_type']]}"] = e[
```
Comment (bmoscon) [inline cryptofeed/pairs.py:229]: undo formatting

File: cryptofeed/rest/binance_futures.py [hunk 4]
```diff
@@ -81,7 +186,8 @@ def _get_trades_hist(self, symbol, start_date, end_date, retry, retry_wait):
         @request_retry(self.ID, retry, retry_wait)
         def helper(start, end):
             if start and end:
-                return requests.get(f"{self.api}aggTrades?symbol={symbol}&limit={REQUEST_LIMIT}&startTime={start}&endTime={end}")
+                return requests.get(
```
Comment (bmoscon) [inline cryptofeed/rest/binance_futures.py:189]: undo formatting

File: cryptofeed/rest/binance_futures.py [hunk 5]
```diff
@@ -105,7 +211,9 @@ def helper(start, end):
                 LOG.warning("%s: No data for range %d - %d", self.ID, start, end)
             else:
                 if data[-1]['T'] == start:
-                    LOG.warning("%s: number of trades exceeds exchange time window, some data will not be retrieved for time %d", self.ID, start)
+                    LOG.warning(
```
Comment (bmoscon) [inline cryptofeed/rest/binance_futures.py:214]: undo formatting

File: cryptofeed/standards.py [hunk 6]
```diff
@@ -76,7 +74,8 @@ def pair_exchange_to_std(pair):
 def timestamp_normalize(exchange, ts):
     if exchange in {BITMEX, COINBASE, HITBTC, OKCOIN, OKEX, BYBIT, FTX, FTX_US, BITCOINCOM, BLOCKCHAIN, PROBIT}:
         return pd.Timestamp(ts).timestamp()
-    elif exchange in {HUOBI, HUOBI_DM, HUOBI_SWAP, BITFINEX, COINBENE, DERIBIT, BINANCE, BINANCE_US, BINANCE_FUTURES, GEMINI, BITTREX, BITMAX, KRAKEN_FUTURES, UPBIT}:
+    elif exchange in {HUOBI, HUOBI_DM, HUOBI_SWAP, BITFINEX, COINBENE, DERIBIT, BINANCE, BINANCE_US, BINANCE_FUTURES,
```
Comment (bmoscon) [inline cryptofeed/standards.py:77]: undo formatting

**Review by bmoscon — CHANGES_REQUESTED** [review]

**Comment by bmoscon** [comment]
the binance delivery code would also need to be in its own file, please change that. If you do not want to make these changes, I can create a new branch with these changes and list you as the co-author of the changeset.

**Comment by bmoscon** [comment]
@txu2014 - thanks for the PR!

**Pull request merged**